pkgs/os-specific/linux/kernel/common-config.nix

```
@@ -846,6 +846,8 @@ let
       MOUSE_PS2_VMMOUSE  = yes;
       MTRR_SANITIZER     = yes;
       NET_FC             = yes; # Fibre Channel driver support
+      # Needed for touchpads to work on some AMD laptops
+      PINCTRL_AMD        = whenAtLeast "5.19" yes;
```

@peterhoeg @lovesegfault It took me quite a while to figure out while my touchpad stopped working after upgrading the `github:NixOS/nixpkgs/nixos-22.05` input of my nixos config flake. It seemed to me that Linux was updated from 5.18.17 to 5.18.19 in the `nixos-22.05` branch, but the changes in #185672 were not yet backported. After more digging (and seeing this PR), it turned out that the issue was commit d0f1798d9b62e91135fbabdddf558e77b6c6e07f. Since the change in the KConfig was [introduced in 5.18.18](https://github.com/gregkh/linux/commits/v5.18.18/drivers/pinctrl/Kconfig), I think the correct change should be:

```suggestion
      PINCTRL_AMD        = whenAtLeast "5.18.18" yes;
```

That said, after browsing the code in `pkgs/os-specific/linux/kernel/` I did not find a 3 component version comparison (even though `builtins.compareVersions` (which is used under the hood) supports them), so I'm hesitant to propose this change.

What do you guys think?

(I can't upgrade to 5.19 since I have `boot.kernelPackages = config.boot.zfs.package.latestCompatibleLinuxPackages` and ZFS does not yet support it.)

---

`whenAtLeast` uses `lib.versionAtLeast` under the hood which will happily take a 3 component version. Feel like whipping up a PR?